name: Continue on error

on:
  pull_request:
    types: [opened, synchronize]

jobs:
  failing_job:
    runs-on: ubuntu-latest
    continue-on-error: true

    steps:
      - name: Fail on purpose
        run: |
          echo "This job is failing..."
          exit 1
      
      - name: Run on fail
        run: |
          echo "This is running in response to a previous step failure..."

  run_on_failure_job:
    runs-on: ubuntu-latest
    needs: [failing_job]
    if: failure()
    steps:
        - name: Acknowledge run
          run: |
            echo "This is job is running due to a previous job failure..."
        
        
